### PR TITLE
Expands on articulation test for multiple instances and devices

### LINKED ISSUE #861: Testing: Improve testing for assets
- Check that actuators for implicit PD and explicit PD are nearly similar
- Check that external forces are applied correctly
- Check that friction/mass/collider offsets randomization works correctly
- Check that apply joint commands work correctly
- Check that setting states work correctly
- Check that reset works correctly

### PR DESCRIPTION
**Pull request by jsmith-bdai**
# Description

Expands upon `test_articulation.py` to test with mulitple robots and on both CPU / GPU.

Fixes #861 

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab/test/assets/test_articulation.py [hunk 1]
```diff
@@ -34,569 +37,758 @@
 from omni.isaac.lab_assets import ANYMAL_C_CFG, FRANKA_PANDA_CFG, SHADOW_HAND_CFG  # isort:skip
 
 
+def generate_articulation_cfg(
+    articulation_type, stiffness: float | None = 10.0, damping: float | None = 2.0
```
Comment (jtigue-bdai) [inline source/extensions/omni.isaac.lab/test/assets/test_articulation.py]: NIT:
```suggestion
    articulation_type: Literal["humanoid", "panda", "anymal", "shandow_hand", "single_joint"], 
    stiffness: float | None = 10.0, 
    damping: float | None = 2.0
```

File: source/extensions/omni.isaac.lab/test/assets/test_articulation.py [hunk 2]
```diff
@@ -34,569 +37,758 @@
...
+) -> ArticulationCfg:
+    """Generate an articulation configuration.
+
+    Args:
+        articulation_type: Type of articulation to generate. Options are "humanoid", "panda", "anymal", "shadow_hand" and "single_joint".
```
Comment (jsmith-bdai) [inline source/extensions/omni.isaac.lab/test/assets/test_articulation.py]:
```suggestion
        articulation_type: Type of articulation to generate.
```

**Comment by Dhoeller19** [comment]
@jsmith-bdai if you look at the AWS log, the test does not run properly:
```  
File "/workspace/isaaclab/source/extensions/omni.isaac.lab/test/assets/test_articulation.py", line 41, in <module>
    articulation_type: Literal["humanoid", "panda", "anymal", "shandow_hand", "single_joint"], 
NameError: name 'Literal' is not defined
```

**Comment by jsmith-bdai** [comment]
> @jsmith-bdai if you look at the AWS log, the test does not run properly:
> 
> ```
> File "/workspace/isaaclab/source/extensions/omni.isaac.lab/test/assets/test_articulation.py", line 41, in <module>
>     articulation_type: Literal["humanoid", "panda", "anymal", "shandow_hand", "single_joint"], 
> NameError: name 'Literal' is not defined
> ```

Ah good catch, I had committed the suggested changes, but haven't re-run yet - will fix.

**Comment by jsmith-bdai** [comment]
@Dhoeller19 @Mayankm96 @kellyguo11 can one of you review this one? It has passed its checks :heavy_check_mark:

**Review by Mayankm96 — APPROVED** [review]

**Pull request merged**